framework/db/QueryBuilder.php

```diff
@@ -151,13 +140,58 @@ public function insert($table, $columns, &$params)
                 $params[$phName] = !is_array($value) && isset($columnSchemas[$name]) ? $columnSchemas[$name]->dbTypecast($value) : $value;
             }
         }
+        return [$names, $placeholders];
+    }
 
-        return 'INSERT INTO ' . $schema->quoteTableName($table)
+    /**
+     * Creates an INSERT SQL statement.
+     * For example,
+     *
+     * ~~~
+     * $sql = $queryBuilder->insert('user', [
+     *  'name' => 'Sam',
+     *  'age' => 30,
+     * ], $params);
+     * ~~~
+     *
+     * The method will properly escape the table and column names.
+     *
+     * @param string $table the table that new rows will be inserted into.
+     * @param array $columns the column data (name => value) to be inserted into the table.
+     * @param array $params the binding parameters that will be generated by this method.
+     * They should be bound to the DB command later.
+     * @return string the INSERT SQL
+     */
+    public function insert($table, $columns, &$params)
+    {
+        list($names, $placeholders) = $this->prepareInsertColumns($table, $columns, $params);
+
+        return 'INSERT INTO ' . $this->db->getSchema()->quoteTableName($table)
             . ' (' . implode(', ', $names) . ') VALUES ('
             . implode(', ', $placeholders) . ')';
     }
 
     /**
+     * Creates an INSERT INTO RETURNING SQL statement or plain INSERT if the underlying DBMS does not support it.
+     * @see insert()
+     *
+     * @param string $table the table that new rows will be inserted into.
+     * @param array $columns the column data (name => value) to be inserted into the table.
+     * @param array $params the binding parameters that will be generated by this method.
+     * They should be bound to the DB command later.
+     * @param array $returnColumns used only when if underlying DBMS support returning data from an INSERT statement
+     * @param array $returnParams used only when if underlying DBMS support returning data from an INSERT statement
+     * They should be bound to the DB command later.
+     * @return string the INSERT SQL
+     */
+    public function insertReturning($table, $columns, &$params, $returnColumns = null, &$returnParams = null)
+    {
+        // clear returnParams
+        $returnParams = null;
```

Comment:
`insertReturning` behaves just like `insert` for DBMS not supporting this. It could throw an exception, but I wanted to avoid that because it would mean that AR would have to catch it each time it saves a record when using MySQL or SQLite, which is very common.

I'm not sure if that's the better solution or an exception should be thrown anyway.